Update `ard_survival_survfit`

edelarua opened the pull request:
**What changes are proposed in this pull request?**
* Updated `ard_survival_survfit` to return `std.error` and `n.risk` statistics. (#139 )

@ddsjoberg as I currently have it, `std.error` gets transformed along with `estimate`, `conf.low`, and `conf.high` (with options `"survival"`, `"cumhaz"`, and `"risk"`), but if you think it's more informative to always return the untransformed `std.error` that can be updated.

Closes #139 

--------------------------------------------------------------------------------

Pre-review Checklist (if item does not apply, mark is as complete)
- [x] **All** GitHub Action workflows pass with a :white_check_mark:
- [x] PR branch has pulled the most recent updates from master branch: `usethis::pr_merge_main()`
- [x] If a bug was fixed, a unit test was added.
- [x] If a new `ard_*()` function was added, it passes the ARD structural checks from `cards::check_ard_structure()`.
- [x] If a new `ard_*()` function was added, `set_cli_abort_call()` has been set.
- [x] If a new `ard_*()` function was added and it depends on another package (such as, `broom`), `is_pkg_installed("broom", reference_pkg = "cardx")` has been set in the function call and the following added to the roxygen comments: `@examplesIf do.call(asNamespace("cardx")$is_pkg_installed, list(pkg = "broom"", reference_pkg = "cardx"))`
- [x] Code coverage is suitable for any new functions/features (generally, 100% coverage for new code): `devtools::test_coverage()`

Reviewer Checklist (if item does not apply, mark is as complete)

- [ ] If a bug was fixed, a unit test was added.
- [ ] Code coverage is suitable for any new functions/features: `devtools::test_coverage()`

When the branch is ready to be merged:
- [ ] Update `NEWS.md` with the changes from this pull request under the heading "`# cardx (development version)`". If there is an issue associated with the pull request, reference it in parentheses at the end update (see `NEWS.md` for examples).
- [ ] **All** GitHub Action workflows pass with a :white_check_mark:
- [ ] Approve Pull Request
- [ ] Merge the PR. Please use "Squash and merge" or "Rebase and merge".

edelarua commented:
I have read the CLA Document and I hereby sign the CLA

ddsjoberg commented:
Thank you @edelarua !!

> @ddsjoberg as I currently have it, std.error gets transformed along with estimate, conf.low, and conf.high (with options "survival", "cumhaz", and "risk"), but if you think it's more informative to always return the untransformed std.error that can be updated.

When we perform a non-linear transformation on a standard error, standard deviation, or variance, we need to be very careful we perform the transformation appropriately (i.e. you could to use the delta method which requires the derivative of the transforming function). Let's keep it simple and not transform the std error, and we'll make it clear with the `stat_label` that this is the std error of the untransformed quantity.  How does that sound?

edelarua commented:
> When we perform a non-linear transformation on a standard error, standard deviation, or variance, we need to be very careful we perform the transformation appropriately (i.e. you could to use the delta method which requires the derivative of the transforming function). Let's keep it simple and not transform the std error, and we'll make it clear with the `stat_label` that this is the std error of the untransformed quantity. How does that sound?

Sounds good @ddsjoberg! I have removed standard error from the transformed variables and changed its label to `Standard Error (untransformed)`.

edelarua commented:
@ddsjoberg this is ready for review when you have the time!!

ddsjoberg reviewed (approved):
Thank you @edelarua !!! this is great
